mssql/compiler.py

```diff
@@ -447,9 +457,9 @@ def as_sql(self):
         qn = self.connection.ops.quote_name
         opts = self.query.get_meta()
         result = ['INSERT INTO %s' % qn(opts.db_table)]
-        fields = self.query.fields or [opts.pk]
 
         if self.query.fields:
+            fields = self.query.fields
```

Comment:
`fields` variable is set here, and also in the `else` below, so the set before `if..else` becomes useless and I've removed it